### PR TITLE
W-14145356: Modularize core-components

### PR DESCRIPTION
**Pull request by mbuchwald**

### CODE REVIEW

**Comment by mbuchwald** [comment]
--validate

**Comment by mbuchwald** [comment]
--validate

File: modules/core-components/pom.xml [hunk 1]
```diff
@@ -86,5 +109,15 @@
             <version>${project.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>net.bytebuddy</groupId>
```
Comment (elrodro83) [inline modules/core-components/pom.xml:113]: isn't this transitive form mockito-inline?

File: modules/core-components/src/main/java/module-info.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  requires org.mule.runtime.tracer.api;
+  requires org.mule.runtime.tracer.customization.api;
+
+  // LoggerMessageProcessor
+  requires org.apache.log4j;
+  // javax.activation
+  requires jakarta.activation;
+  // QName
+  requires java.xml;
+  requires java.desktop;
```
Comment (elrodro83) [inline modules/core-components/src/main/java/module-info.java:28]: what is used form this/who uses this?

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by elrodro83 — APPROVED** [review]

**Comment by mbuchwald** [comment]
--validate

**Comment by mbuchwald** [comment]
--validate

**Comment by mbuchwald** [comment]
--validate

**Comment by mbuchwald** [comment]
--validate

**Comment by mbuchwald** [comment]
--validate

**Comment by mbuchwald** [comment]
--validate

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**